_posts/2019-07-19-paho-mqtt.md

```diff
@@ -0,0 +1,114 @@
+
+You will need two devices, either a Debian machine or Raspberry Pi. In a set up where you have both, it is preferable to set up the Debian machine as your server device. Both these devices will need to have the [WoTT Agent]({{site.url}}/documentation/getting-started) installed and [Docker CE](https://docs.docker.com/install/linux/docker-ce/debian/) installed. Note installation instructions for Docker on Pi is different so refer to the link above for appropriate instructions. 
+
+This example contains code snippets. We recommend you clone our [examples](https://github.com/WoTTsecurity/examples) repository by doing the following:
+
+```
+$ git clone https://github.com/WoTTsecurity/examples.git
+```
+Alternatively, you can use `curl` to manually install the files that you need. Refer to our other code snippet examples ([Adafruit IO]({{site.url}}/blog/tutorials/2019/06/27/adafruit-io), [Simple WebApp]({{site.url}}/blog/tutorials/2019/06/16/simple-webapp), etc) for instructions to do this. However, if you are going to be looking at our other tutorials in the future we recommend you have the repository readily available.
+
+## Setting up the server (Mosquitto)
+
+Please refer to our [Mosquitto MQTT example](({{site.url}}/blog/tutorials/2019/07/15/mosquitto-mqtt)) for in depth information on how to set up the server. If you are familiar with this example and want to just quickly 'set up and go,' do the following:
+ 
+Navigate to the directory where you have cloned our 'examples' repository:
+
+```
+$ cd examples
+$ cd mosquitto-server
+```
+
+Alter the ACL file `permissions.acl` with your WoTT device details for the relevant permissions (remember we want to grant some access to your client device). To find the WoTT Device ID:
+
+```
+$ sudo wott-agent whoami
+```
+
+We are still using the topic `/wott/temperature` to publish and subscribe messages to and from.
+
+## Setting up the client (Paho)
+
+This is where our instructions differ from setting up the Mosquitto client. It is not significantly more complicated but Paho has a different set up. First, you will need to have Paho on your system.
+Navigate to the directory containing our `examples` respository. Navigate to the `paho-mqtt-client` folder like so:
+
+```
+$ cd examples
+$ cd paho-mqtt-client
+$ pip3 install -r requirements.txt
+```
+
+There are two files in this folder - a `requirements.txt` file and the client itself `paho-connect-wott.py`. The above command installs the version of the Paho client needed for this example. 
+
+**Note:** you can use whatever version of the `pip` tool you have.
+
+Now that Paho is installed, open the `paho-connect-wott.py` file in your favourite text editor. Towards the end of the document (line 33) you will notice this line:
+
+```
+client.connect('ServerDeviceID.d.wott.local', 8883, 60)
```

Comment:
Let's remove this. I've pushed an update to the code and changed it such that the user won't have to edit the code base (which makes things messy otherwise).

So, instead of editing the code, we'll rely on environment variables.

```
$ export MQTT_SERVER=x.d.wott.local

# This defaults to "wott/temperature"
$ export MQTT_TOPIC=my/topic

# Only if the port is different than 8333
$ export MQTT_PORT=8333 
```

Reply:
so if the the person does `$ export MQTT_TOPIC=my/topic` verbatim, the code will automatically infer that as `wott/temperature` ?

Reply:
Correct.